packages/inter-protocol/test/swingsetTests/psmUpgrade/test-psm-upgrade.js

```diff
@@ -0,0 +1,90 @@
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+
+import { assert } from '@agoric/assert';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { resolve as importMetaResolve } from 'import-meta-resolve';
+
+import { buildVatController } from '@agoric/swingset-vat';
+import { faV1BundleName } from './bootstrap-psm-upgrade.js';
+
+// so paths can be expresssed relative to this file and made absolute
+const bfile = name => new URL(name, import.meta.url).pathname;
+
+test('PSM service upgrade', async t => {
```

Comment:
These upgrade tests are so verbose that it's a little hard to see the forest for the trees. But it's not obvious how to do much better.

To confirm: this tests that before and after a null upgrade, we can do PSM trades, yes? If there's a difference between `testFunctionality1` and `testFunctionality2`, I don't see it.